lint: paramTypeCombine gives false invalid suggestion for unnamed params

Reproducer:
```
gocritic check-package -enable=paramTypeCombine bytes
```

Output:
```
src/bytes/bytes.go:946:1: paramTypeCombine: func(sep []byte) (uint32, uint32) could be replaced with func(sep []byte) (uint32)
```
Function:
```go
func hashStr(sep []byte) (uint32, uint32) {
```
The `(uint32, uint32)` != `(uint32)`.

paramTypeCombine should probably skip this case and do not emit warning.
Note that `unnamedResult` will complain about that:
```
bytes/bytes.go:946:1: unnamedResult: consider to give name to results
```
And if advice is followed, new issue can be found by `paramTypeCombine`.

I'm almost sure `paramTypeCombine` should not suggest giving names to types just to combine them,
there is also `unusedParam` checker that suggests to use `_` for every unused param, so there are quite a bit effort around that already.

--- a/cmd/gocritic/testdata/paramTypeCombine/negative_tests.go
+++ b/cmd/gocritic/testdata/paramTypeCombine/negative_tests.go
@@ -8,3 +8,9 @@ func good5(a int)                   {}
 func good6()                        {}
 
 func mixedTypes(a, b int, c, d int64) {}
+
+func unnamedParams(uint32, uint32) {}
+
+func unnamedResults(sep []byte) (uint32, uint32) {
+	return 0, 0
+}

--- a/cmd/gocritic/testdata/paramTypeCombine/positive_tests.go
+++ b/cmd/gocritic/testdata/paramTypeCombine/positive_tests.go
@@ -35,3 +35,9 @@ func mixedStyle8(a int, b, c int) (d int, e int) { return a, c }
 
 /// func(a int, b int, c int64, d int, e, f int64, _, g int64, h int, k int) could be replaced with func(a, b int, c int64, d int, e, f, _, g int64, h, k int)
 func mixedTypeWarn(a int, b int, c int64, d int, e, f int64, _, g int64, h int, k int) {}
+
+/// func(_, _ int, _ int, _ int32) could be replaced with func(_, _, _ int, _ int32)
+func withBlank1(_, _ int, _ int, _ int32) {}
+
+/// func() (_, _ int, _ int, _ int32) could be replaced with func() (_, _, _ int, _ int32)
+func withBlank2() (_, _ int, _ int, _ int32) { return }

--- a/lint/paramTypeCombine_checker.go
+++ b/lint/paramTypeCombine_checker.go
@@ -36,7 +36,15 @@ func (c *paramTypeCombineChecker) optimizeFuncType(f *ast.FuncType) *ast.FuncTyp
 	}
 }
 func (c *paramTypeCombineChecker) optimizeParams(params *ast.FieldList) *ast.FieldList {
-	if params == nil || len(params.List) < 2 {
+	// To avoid false positives, skip unnamed param lists.
+	//
+	// We're using a property that Go only permits unnamed params
+	// for the whole list, so it's enough to check whether any of
+	// ast.Field have empty name list.
+	skip := params == nil ||
+		len(params.List) < 2 ||
+		len(params.List[0].Names) == 0
+	if skip {
 		return params
 	}
 

